.github/pull_request_template.md

```diff
@@ -1,32 +1,39 @@
 <!--
 
-Describe in detail the changes you are proposing, and the rationale.
+** Thank you for your contribution! Please read this carefully! **
 
-See the contributing guide:
-
-https://github.com/opentofu/opentofu/blob/main/CONTRIBUTING.md
+Please make sure you go through the checklist below. If your PR does not meet all requirements, please file it
+as a draft PR. Core team members will only review your PR once it meets all the requirements below (unless your
+change is something as trivial as a typo fix).
 
 -->
 
-<!--
+<!-- If your PR resolves an issue, please add it here. -->
+Resolves # 
 
-Link all GitHub issues fixed by this PR, and add references to prior related PRs.
-Make sure to first open an issue, get community approval and only then create Pull Request to resolve it.
-All Pull Requests must have an issue attached to them
+## Target Release
 
--->
+1.7.0
 
-Resolves #
+## Checklist
 
-## Target Release
+<!-- This checklist is mandatory for all PRs: -->
 
-<!--
+- [ ] I have read the [contribution guide](https://github.com/opentofu/opentofu/blob/main/CONTRIBUTING.md).
+- [ ] I have not used an AI coding assistant to create this PR.
+- [ ] I have marked any code I have not written myself (including modified code, e.g. copied from other places and then modified) with a comment indicating where it came from. 
 
-In normal circumstances we only target changes at the upcoming minor
-release, or as a patch to the current minor version. If you need to
-port a security fix to an older release, highlight this here by listing
-all targeted releases.
+### Go checklist
 
--->
+<-- If your PR contains Go code, please make sure you check off all items on this list: --> 
 
-1.7.0
+- [ ] I have run golangci-lint on my change and receive no errors relevant to my code.
```

Comment:
I think adding golangci-lint check to the github action workflow will be better.

If it is already exist, ignore me.

Reply:
I think we need both local and remote lint checks. Also, it would be useful to have a script running as a pre-commit hook that runs lint, tests, and verifies the commit is going to have a sign-off.

Reply:
One does not preclude the other. However, contributors should be aware that their PR will not be reviewed in-depth until they pass the basic check list and we'll want to keep the PRs in draft state so we know what is ready for review and what isn't.

Reply:
I will raise a separate issue to run linting as an automated process (pre-commit or github actions or both)